namer/consul/src/main/scala/io/buoyant/namer/consul/SvcAddr.scala

```diff
@@ -46,20 +49,30 @@ private[consul] object SvcAddr {
     preferServiceAddress: Option[Boolean] = None,
     tagWeights: Map[String, Double] = Map.empty,
     stats: Stats,
-    stateWatch: v1.PollState[String, v1.IndexedServiceNodes]
+    stateWatch: v1.PollState[String, v1.IndexedServiceNodes],
+    transferMetaData: Option[Boolean] = None
   )(implicit timer: Timer = DefaultTimer): InstrumentedVar[Addr] = {
     val meta = mkMeta(key, datacenter, domain)
-    def getAddresses(index: Option[String]): Future[v1.Indexed[Set[Address]]] = {
+
+    def getAddresses(index: Option[String]): Future[v1.Indexed[MetaWithAddresses]] = {
       val apiCall = consulApi.serviceNodes(
         key.name,
         datacenter = Some(datacenter),
         tag = key.tag,
         blockingIndex = index,
-        consistency = consistency,
-        retry = false
+        consistency = consistency
       )
       v1.InstrumentedApiCall.execute(apiCall, stateWatch)
-        .map(indexedToAddresses(preferServiceAddress, tagWeights))
+        .map {
+          case v1.Indexed(nodes, idx) => {
+            val meta: v1.MetaData = nodes.headOption.flatMap(_.ServiceMeta) -> transferMetaData match {
```

Comment:
The `headOption` confused me at first.  It might be good to add a comment explaining that all nodes are assumed to have the same ServiceMeta.